Src/Newtonsoft.Json-for-Unity.Tests/Assets/SystemWebPolyfill/Util/HttpEncoder.cs

```diff
@@ -0,0 +1,796 @@
+
+        protected internal virtual byte[] UrlEncode (byte[] bytes, int offset, int count)
+        {
+            return UrlEncodeToBytes (bytes, offset, count);
+        }
+
+        static HttpEncoder GetCustomEncoderFromConfig ()
+        {
+#if MOBILE || NO_SYSTEM_WEB_DEPENDENCY || true
+            return defaultEncoder.Value;
+#else
+            var cfg = HttpRuntime.Section;
+            string typeName = cfg.EncoderType;
+
+            if (String.Compare (typeName, "System.Web.Util.HttpEncoder", StringComparison.OrdinalIgnoreCase) == 0)
+                return Default;
+
+            Type t = Type.GetType (typeName, false);
+            if (t == null)
+                throw new ConfigurationErrorsException (String.Format ("Could not load type '{0}'.", typeName));
+
+            if (!typeof (HttpEncoder).IsAssignableFrom (t))
+                throw new ConfigurationErrorsException (
+                    String.Format ("'{0}' is not allowed here because it does not extend class 'System.Web.Util.HttpEncoder'.", typeName)
+                );
+
+            return Activator.CreateInstance (t, false) as HttpEncoder;
+#endif
+        }
+        protected internal virtual
+        string UrlPathEncode (string value)
+        {
+            if (String.IsNullOrEmpty (value))
+                return value;
+
+            MemoryStream result = new MemoryStream ();
+            int length = value.Length;
+            for (int i = 0; i < length; i++)
+                UrlPathEncodeChar (value [i], result);
+
+            return Encoding.ASCII.GetString (result.ToArray ());
+        }
+
+        internal static byte[] UrlEncodeToBytes (byte[] bytes, int offset, int count)
+        {
+            if (bytes == null)
+                throw new ArgumentNullException ("bytes");
+
+            int blen = bytes.Length;
+            if (blen == 0)
+                return new byte [0];
+
+            if (offset < 0 || offset >= blen)
+                throw new ArgumentOutOfRangeException("offset");
+
+            if (count < 0 || count > blen - offset)
+                throw new ArgumentOutOfRangeException("count");
+
+            MemoryStream result = new MemoryStream (count);
+            int end = offset + count;
+            for (int i = offset; i < end; i++)
+                UrlEncodeChar ((char)bytes [i], result, false);
+
+            return result.ToArray();
+        }
+
+        internal static string HtmlEncode (string s)
+        {
+            if (s == null)
+                return null;
+
+            if (s.Length == 0)
+                return String.Empty;
+
+            bool needEncode = false;
+            for (int i = 0; i < s.Length; i++) {
+                char c = s [i];
+                if (c == '&' || c == '"' || c == '<' || c == '>' || c > 159
+                    || c == '\''
+                ) {
+                    needEncode = true;
+                    break;
+                }
+            }
+
+            if (!needEncode)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity/pullRequest?prid=4755183)